index.js

```diff
@@ -51,6 +57,7 @@ function init() {
     ui.metricPerTile = metricPerTile
     canvas.updateTilesFromMetrics(metricPerTile, sumMetrics)
   })
+  ui.setUnsavedChangesCallback(() => canvas.getGrid().checkForEdits())
```

Comment:
Hmm, I'm not sure you do need one here. in `checkForEdits` you access `_hasBeenEdited` of the `GridGraphic` not of `ui`. I don't think either of the callbacks use `this` they just call other functions.